UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -949,25 +952,77 @@ protected virtual async Task BuildAnimationSamplers(GLTFAnimation animation, int
 			{
 				var time = input.AsFloats[i];
 
-				var values = getConvertedValues(output, i);
+				float[] values = null;
+				float[] inTangents = null;
+				float[] outTangents = null;
+				if (mode == InterpolationType.CUBICSPLINE)
+				{
+					// For cubic spline, the output will contain 3 values per keyframe; inTangent, dataPoint, and outTangent.
+					// https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#appendix-c-spline-interpolation
+
+					var cubicIndex = i * 3;
+					inTangents = getConvertedValues(output, cubicIndex);
+					values = getConvertedValues(output, cubicIndex + 1);
+					outTangents = getConvertedValues(output, cubicIndex + 2);
+				}
+				else
```

Comment:
suggest adding the full condition if(interpolationmode =keyframe ||  something else )  in this case to make it clear what type of interpolation that goes here